app/src/main/java/com/example/pinterbest/di/ApplicationComponent.kt

```diff
@@ -0,0 +1,22 @@
+package com.example.pinterbest.di
+
+import android.content.Context
+import com.example.pinterbest.presentation.common.ApplicationAPI
+import dagger.BindsInstance
+import dagger.Component
+import javax.inject.Singleton
+
+@Singleton
+@Component(
+    modules = [
+        AppModule::class, PinsModule::class,
+        ViewModelModule::class, AuthModule::class, ProfileModule::class, SessionModule::class
```

Comment:
nit: this may be cleaner if rewrite each module on new line, like

```kotlin
modules = [
  AppModule::class,
  PinsModule::class,
  ViewModelModule::class,
  AuthModule::class,
  ProfileModule::class,
  SessionModule::class,
```

Reply:
Fixed